### PR TITLE
Rendering system yas

### PR DESCRIPTION
**Pull request by ybayoumy**
- Added Mesh/Model classes to represent geometry
- Added basic Camera class
- now using indexed geometry
- Added RenderingSystem class. Supports loading Models from files (not tested yet)

### CODE REVIEW

**Review by RatikKapoor — APPROVED** [review]
Well done! Works really well for me! :shipit: 

Just missing some docs on the addition of `assimp:x64-windows-static`, I'd be okay with including these in a separate PR to unblock other work!

**Comment by RatikKapoor** [comment]
Pending review from @brooke-k for merge!

**Review by brooke-k — APPROVED** [review]
The triangle falls. Excellent. 🔺

**Pull request merged**
